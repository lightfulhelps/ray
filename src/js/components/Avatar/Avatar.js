// @flow
import React from 'react';
import classNames from 'classnames';
import { Icon } from '../../';

type Props = {
  className?: string,
  provider?: 'facebook' | 'twitter' | 'linkedin',
  url?: string,
};

const Avatar = ({ className, url, provider }: Props) => {
  const classes = classNames(className, 'avatar');

  return (
    <div className={classes}>
      <div className="avatar-image-wrap">
        {url && <img className="avatar-image" src={url} alt="" />}
      </div>
      {provider && (
        <div className={`avatar-provider avatar-provider-${provider}`}>
          <Icon name={provider} color="#ffffff" size={16} />
        </div>
      )}
    </div>
  );
};

export default Avatar;
