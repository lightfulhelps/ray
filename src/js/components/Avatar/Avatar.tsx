
import React from "react";
import classNames from "classnames";
import { Icon } from "../../";

type Props = {
  className?: string;
  isDisconnected?: boolean;
  provider?: "facebook" | "twitter" | "linkedin";
  providerSize?: "sm" | "md" | "lg";
  url?: string;
};

const Avatar = ({
  className,
  isDisconnected,
  url = '',
  provider,
  providerSize = 'md',
  ...other
}: Props) => {
  const classes = classNames(className, 'avatar', `avatar-provider-${providerSize}`, {
    'avatar-disconnected': isDisconnected
  });

  return <div {...other} className={classes}>
      <div className="avatar-inner">
        <div className="avatar-image" style={{ backgroundImage: `url(${url})` }} />
        {provider && <div className={`avatar-provider avatar-provider-${provider}`}>
            <Icon name={provider} color="#ffffff" />
          </div>}
      </div>
    </div>;
};

export default Avatar;