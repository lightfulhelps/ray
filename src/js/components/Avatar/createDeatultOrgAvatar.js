const createDeatultOrgAvatar = orgName => {
  const letter = orgName ? orgName[0] : ' ';
  let bgClass;
  if (/[A-F]/.test(letter)) {
    bgClass = 'bg-secondary';
  }
  if (/[G-L]/.test(letter)) {
    bgClass = 'bg-primary';
  }
  if (/[M-R]/.test(letter)) {
    bgClass = 'bg-danger';
  }
  if (/[S-Z]/.test(letter)) {
    bgClass = 'bg-warning';
  }
  if (!/[A-Z]/.test(letter)) {
    bgClass = 'bg-info';
  }
  return { bgClass, letter };
};

export default createDeatultOrgAvatar;
