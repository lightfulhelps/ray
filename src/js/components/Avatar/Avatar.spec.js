import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';
import Avatar from './Avatar';
import { Icon } from '../../';

const setup = (overrides = {}) => {
  const props = merge(
    {
      url: 'https://randomuser.me/api/portraits/women/47.jpg',
      provider: 'twitter',
    },
    overrides
  );
  const wrapper = shallow(<Avatar {...props} />);

  return {
    wrapper,
    props,
  };
};

describe('<Avatar />', () => {
  it('should render', () => {
    const { wrapper } = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it('should handle the className prop', () => {
    const { wrapper } = setup({ className: 'custom' });

    expect(wrapper.hasClass('avatar')).toBe(true);
    expect(wrapper.hasClass('custom')).toBe(true);
  });

  it('should handle the url prop', () => {
    const { wrapper } = setup({ url: null });

    expect(wrapper.find('.avatar-image').exists()).toBe(false);

    wrapper.setProps({ url: 'https://randomuser.me/api/portraits/women/47.jpg' });

    expect(wrapper.find('.avatar-image').exists()).toBe(true);
    expect(wrapper.find('.avatar-image').prop('src')).toEqual(
      'https://randomuser.me/api/portraits/women/47.jpg'
    );
  });

  it('should handle the provider prop', () => {
    const { wrapper } = setup({ provider: null });

    expect(wrapper.find('.avatar-provider').exists()).toBe(false);

    wrapper.setProps({ provider: 'facebook' });

    expect(wrapper.find('.avatar-provider').exists()).toBe(true);
    expect(
      wrapper
        .find('.avatar-provider')
        .find(Icon)
        .prop('name')
    ).toEqual('facebook');
  });
});
