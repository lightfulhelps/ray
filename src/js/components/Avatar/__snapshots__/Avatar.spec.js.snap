// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> should render 1`] = `
<div
  className="avatar"
>
  <div
    className="avatar-image-wrap"
  >
    <img
      alt=""
      className="avatar-image"
      src="https://randomuser.me/api/portraits/women/47.jpg"
    />
  </div>
  <div
    className="avatar-provider avatar-provider-twitter"
  >
    <Icon
      color="#ffffff"
      name="twitter"
    />
  </div>
</div>
`;
